Frontend/ui-customer/src/Flow.purs

```diff
@@ -576,17 +576,25 @@ homeScreenFlow = do
       setValueToLocalStore TRACKING_ID (getNewTrackingId unit)
       when (getValueToLocalStore FLOW_WITHOUT_OFFERS == "true") do
         void $ pure $ firebaseLogEvent "ny_user_auto_confirm"
-      void $ pure $ setValueToLocalStore FINDING_QUOTES_START_TIME (getCurrentUTC "LazyCheck")
 
-      response <- lift $ lift $ Remote.selectEstimate (Remote.makeEstimateSelectReq (flowWithoutOffers WithoutOffers) (if state.props.customerTip.enableTips && state.props.customerTip.isTipSelected then Just state.props.customerTip.tipForDriver else Nothing)) (state.props.estimateId)
-      case response of
-        Right res -> do
-          updateLocalStage FindingQuotes
-          modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{ props { currentStage = FindingQuotes, searchExpire = (getSearchExpiryTime "LazyCheck") } })
-        Left err -> do
-          void $ pure $ firebaseLogEvent "ny_user_estimate_expired"
-          updateLocalStage FindEstimateAndSearch
-          modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{ props { currentStage = FindEstimateAndSearch } })
+      let currentTime = (convertUTCtoISC (getCurrentUTC "") "HH:mm:ss")
+          findingQuotesTime = convertUTCtoISC (getValueToLocalNativeStore FINDING_QUOTES_START_TIME) "HH:mm:ss"
+      if withinTimeRange findingQuotesTime currentTime "22:00:00" || withinTimeRange findingQuotesTime currentTime "05:00:00" then do
+        void $ pure $ toast (getString PLEASE_FIND_REVISED_FARE_ESTIMATE)
+        void $ pure $ firebaseLogEvent "ny_user_new_estimate_after_night_charges_applicable"
+        updateLocalStage FindEstimateAndSearch
+        modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{ props { currentStage = FindEstimateAndSearch, searchAfterEstimate = false } })
+      else do
+        void $ pure $ setValueToLocalStore FINDING_QUOTES_START_TIME (getCurrentUTC "LazyCheck")
+        response <- lift $ lift $ Remote.selectEstimate (Remote.makeEstimateSelectReq (flowWithoutOffers WithoutOffers) (if state.props.customerTip.enableTips && state.props.customerTip.isTipSelected then Just state.props.customerTip.tipForDriver else Nothing)) (state.props.estimateId)
+        case response of
+          Right res -> do
+            updateLocalStage FindingQuotes
+            modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{ props { currentStage = FindingQuotes, searchExpire = (getSearchExpiryTime "LazyCheck") } })
```

Comment:
@ashishsingh101  can we use the LazyCheck constructor itself instead of sending a string

Reply:
its none of the mentioned data type